adapters/repos/db/inverted/new_prop_length_tracker.go

```diff
@@ -256,29 +324,50 @@ func (t *JsonPropertyLengthTracker) PropertyTally(propName string) (int, int, fl
 	return sum, count, float64(sum) / float64(count), nil
 }
 
-// Writes the current state of the tracker to disk.  (flushBackup = true) will only write the backup file
-func (t *JsonPropertyLengthTracker) Flush(flushBackup bool) error {
-	if !flushBackup { // Write the backup file first
-		t.Flush(true)
+// Returns the number of documents stored in the shard
+func (t *JsonShardMetaData) ObjectTally() int {
+	if t == nil {
+		return 0
 	}
 
 	t.Lock()
 	defer t.Unlock()
 
+	return t.data.ObjectCount
+}
+
+func (t *JsonShardMetaData) Flush() error {
+	if t == nil {
+		return nil
+	}
+
+	t.Lock()
+	defer t.Unlock()
+
+	return t.lockFreeFlush()
+}
+
+func (t *JsonShardMetaData) lockFreeFlush() error {
+	t.WantFlush = true // Remove this line once we add delayed flushing
```

Comment:
I'm not sure this is necessary, all the places that set WantFlush to false are behind a lock.

Reply:
WantFlush isn't about locks, it's the deferral mechanism

So ideally only flush sets wantflush to false, then the next time you change a property it gets set to true, then 5 secs later it gets written.

That part didn't work and isn't included in this patch

Reply:
So I should take it out, one sec.